protected/humhub/modules/space/views/space/_layout.php

```diff
@@ -24,7 +25,12 @@
     </div>
     <div class="row space-content">
         <div class="col-md-2 layout-nav-container">
-            <?= Menu::widget(['space' => $space]); ?>
+            <?php
+            if ($space->isMember())
+                echo Menu::widget(['space' => $space]);
+            else
+                echo NonMembersMenu::widget(['space' => $space]); //for stream page
```

Comment:
Why do we need an extra NonMemberMenu widget here?

Reply:
At the first, I created this widget to omit modules items that are added to menus, but yes we can remove it and just check if a user is not a member and unset modules items on the menu.

Reply:
Ah ok.

In case someone is invited to a private (invisible) space and is not yet a member, I would simply hide the navigation or delete all items except the about page.

For all other cases, the modules should handle the IsMember/NoMember cases and only inject to the menu if useful.